### ISSUE TITLE
Tests failing when extending AppCompatActivity/using AppCompatDelegate in Activity

### ISSUE THREAD

**Issue by eduplus** [robolectric/robolectric#1846]
android.support.v7.internal.widget.NativeActionModeAwareLayout cannot be cast to android.support.v7.internal.widget.ContentFrameLayout
java.lang.ClassCastException: android.support.v7.internal.widget.NativeActionModeAwareLayout cannot be cast to android.support.v7.internal.widget.ContentFrameLayout
	at android.support.v7.app.AppCompatDelegateImplV7.ensureSubDecor(AppCompatDelegateImplV7.java:372)
	at android.support.v7.app.AppCompatDelegateImplV7.setContentView(AppCompatDelegateImplV7.java:246)

This is the line of code that results for that stacktrace:

    @Override
    public void setContentView(@LayoutRes int layoutResID) {
        getDelegate().setContentView(layoutResID);
    }

Tests were working before starting to use AppCompatDelegate. AppCompatDelegate is used the same way AppCompatActivity is using it.

**Comment by eduplus** [comment]
This seems to happen in fact if Activity extends AppCompatActivity as well, which makes Activities unusable if used with the new support library. For us this is all activities, so it makes robolectric unusable at this time.

**Comment by eduplus** [comment]
After finding out that there is rc3 I tested running with that and it seems it goes further, but still has an exception:

```
android.support.v4.view.ViewCompat.isLaidOut(Landroid/view/View;)Z
java.lang.NoSuchMethodError: android.support.v4.view.ViewCompat.isLaidOut(Landroid/view/View;)Z
	at android.support.v7.internal.widget.ContentFrameLayout.setDecorPadding(ContentFrameLayout.java:73)
	at android.support.v7.app.AppCompatDelegateImplV7.applyFixedSizeWindow(AppCompatDelegateImplV7.java:431)
	at android.support.v7.app.AppCompatDelegateImplV7.ensureSubDecor(AppCompatDelegateImplV7.java:406)
	at android.support.v7.app.AppCompatDelegateImplV7.setContentView(AppCompatDelegateImplV7.java:246)
	at com.irofit.ziroo.android.activity.LoginActivity.setContentView(LoginActivity.java:142)
```

**Comment by erd** [comment]
Closing this in favor of #1859, which is also related to supporting `AppCompatActivity`.
